pkg/storage/admitters/vmrestore.go

```diff
@@ -251,3 +259,50 @@ func (admitter *VMRestoreAdmitter) validatePatches(patches []string, field *k8sf
 
 	return causes
 }
+
+func (admitter *VMRestoreAdmitter) validateVolumeOverrides(ctx context.Context, vmRestore *snapshotv1.VirtualMachineRestore) (causes []metav1.StatusCause, err error) {
+	// Cancel if there's no volume override
+	if vmRestore.Spec.VolumeRestoreOverrides == nil {
+		return nil, nil
+	}
+
+	// Check each individual override
+	for i, override := range vmRestore.Spec.VolumeRestoreOverrides {
+		namespace := vmRestore.Namespace
+		vmSnapshot, err := admitter.Client.VirtualMachineSnapshot(namespace).Get(ctx, vmRestore.Spec.VirtualMachineSnapshotName, metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil, nil
+			}
+			return nil, err
+		}
+
+		contentName := vmSnapshot.Status.VirtualMachineSnapshotContentName
+		if contentName == nil {
+			return nil, fmt.Errorf("snapshot content name is nil in vmSnapshot status")
+		}
+
+		vmSnapshotContent, err := admitter.Client.VirtualMachineSnapshotContent(namespace).Get(ctx, *contentName, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		var found bool
+		for _, vol := range vmSnapshotContent.Spec.VolumeBackups {
+			if vol.VolumeName == override.VolumeName {
+				found = true
+				break
+			}
+		}
+
+		if !found {
```

Comment:
Not sure we want to do this validation given that VMRestore can be created before a VMSnapshot even exists. This check could fail mid snapshot for a legit configuration when VMSnapshot exists but VMSnapshotContent does not.

In webhooks, we generally we try to avoid referencing any resources other than the one being validated unless absolutely necessary. This has been a "lesson learned" in the project. Initially we thought we were being super helpful to the user by doing all sorts of checks in webhooks but that strategy falls apart in an eventually consistent world